### ISSUE TITLE
Playback timing issue with some ogg files

### ISSUE THREAD

**Issue by mikeoliphant** [NVorbis/NVorbis#13]
I recently upgraded to the latest NAudio/NVorbis and I am now getting some playback synchronization issues with files that used to work.

Some files when decoded get slightly out of sync over time compared to how they should be.

Here is an example file:

https://drive.google.com/file/d/1KnMiL29oW8x9LA1xQtLAj9347AWJdKWt/view?usp=sharing

Here is that ogg decoded to a .wav file using NVorbis:

https://drive.google.com/file/d/1BXXs2OGqWHoqrguVJtlByb2r7_dH9dDg/view?usp=sharing

If you load both the .ogg file and the .wav file into an audio editor (such as Audacity), you will easily be able to see/hear the progressive sync issue.

The .wav was created from the .ogg as follows:

`using (var fs = File.OpenRead(fileName))
{
    using (var waveStream = new VorbisWaveStream(fs))
    {
        WaveFileWriter.CreateWaveFile16(Path.Combine(@"C:\tmp", Path.GetFileNameWithoutExtension(fileName) + ".wav"), waveStream);
    }
}
`

As I said, this file (and other files I have issues with) worked fine previously. I *think* I was previously using NAudio.Vorbis 1.2.0.

**Comment by mikeoliphant** [comment]
Some more information on this. When decoding the ogg file I linked above, it reports that it has 14436798 samples. The decoded wav file only ends up with 14407104 samples, though, so somehow samples are getting dropped.

Loading the ogg file directly into Audacity shows the expected 14436798 sample length.

**Comment by mikeoliphant** [comment]
I did some testing with older versions. Decoding works fine in v0.9.1 and the problem occurs in v0.10.0-alpha. So it looks like something in the 0.10 rewrite caused the problem. Unfortunately with all of the changes in that version it may be hard to track down.

**Comment by ioctlLR** [comment]
Yeah, v0.10 has a completely new implementation of the Ogg format reader that addresses performance and memory usage issues previous versions had.  Unfortunately it appears to have a few issues.

We've been snowed-in for the last week, but lemme see if I can find some time to take a look at this later today or tomorrow.

**Comment by mikeoliphant** [comment]
Let me know if there is anything I can do to help, or if you need any other files that show this behavior.

**Comment by mikeoliphant** [comment]
Thanks - good to see this fixed!
